src/plugins/presentation_util/public/components/solution_toolbar/solution_toolbar.tsx

```diff
@@ -66,6 +68,7 @@ export const SolutionToolbar = ({ isDarkModeEnabled, children }: Props) => {
           ) : null}
         </EuiFlexGroup>
       </EuiFlexItem>
+      {controlsButton ? <EuiFlexItem grow={false}>{controlsButton}</EuiFlexItem> : null}
```

Comment:
I discussed with @Heenawter about passing this `controlsButton` as part of `extraButtons` instead of introducing a new prop specifically for controls, but current the `extraButtons` are rendered between the quick buttons and the add from library button. We want the controls button to appear after the `addFromLibrary` button to keep the visualization/panel buttons separate from the controls buttons. 

This lead to a different discussion about supporting a way for the consumer of `SolutionToolbar` to have control over the ordering of the additional buttons on the toolbar to the right of the `quickButtons`. 

We decided one way we could handle this would be to remove the `addFromLibrary` prop altogether and just pass your `addFromLibrary` button in to `extraButtons`, so have something like 
```js
extraButtons=[button1, addFromLibraryButton, button2, button3]
```

@elastic/shared-ux I know you're currently working on extracting out the solution toolbar component into the shared-ux plugin. What do you think about this solution?

Reply:
Depending on what @elastic/shared-ux thinks about this change, I will implement this in this PR! The current code adds the controls button as a new prop to `SolutionToolbar` but this is probably not sustainable, especially if we plan to start using `SolutionToolbar` in places other than dashboard - definitely needs to be explored further :)

Reply:
Very much agreed with the discussion. The more generic this toolbar can become the better, especially if it'll be moved to shared-ux.